chef/automate
# Jamie/natural sort library function 2

### :nut_and_bolt: Description: What code changed, and why?
It was determined that it might be useful if we created a helper function for Normal Sorting, which is what we use in our projects-dropdown component.  This could then be reused across our codebase where necessary.

A sorter file was created in app/helpers/auth with a ChefSorters class to hold any future sorting functions we might want to build.

Components refactored to use ChefSorters helper class:
- [x] project-dropdown.component.ts
- [x] api-token-list.component.js
- [x] policy-add-members.component.ts
- [x] policy-list.component.js
- [x] project-list.component.js
- [x] projects-filter.reducer.ts
- [x] roles-list.component.js
- [x] team-add-users.component.js
- [x] team-details.component.js
- [x] team-management.component.js
- [x] user-management.component.js

### :chains: Related Resources
https://github.com/chef/automate/pull/1620
[First PR where rebase mistake was made](https://github.com/chef/automate/pull/1639)

### :+1: Definition of Done
A more universal version of a Normal Sorting function is available to use across the codebase, and all components that current use natural sorting in this manner are updated.

### :athletic_shoe: How to Build and Test the Change

Visually -
1. build components/automate-ui-devproxy && start_all_services
2. Navigate to https://a2-dev.test/settings/tokens/ <your-api-token>
3. Click on the projects dropdown to see your projects sorted normally.
4. Navigate to https://a2-dev.test/settings/tokens --> View tokens sorted by Name
5. Navigate to https://a2-dev.test/settings/policies --> View policy list sorted by Name
6. From policies, click on Editors then Members Tab.  click Add Member --> View Members list sorted by Display Name first, then Name if possible.
7.  Navigate to https://a2-dev.test/settings/projects --> View projects sorted by Name
8. Navigate to https://a2-dev.test/dashboards/event-feed. click on "All Projects" in top right corner --> view Projects sorted by name, with (Unassigned) as last.
9. Navigate to https://a2-dev.test/settings/roles --> view roles list sorted by Name
11. Navigate to https://a2-dev.test/settings/teams/admins --> see list sorted by Name first, then by id if necessary.
10. Click "Add User". --> View list sorted by Name first, then by id if necessary.
11. Navigate to https://a2-dev.test/settings/teams ---> view teams sorted by ID
12. Navigate to https://a2-dev.test/settings/users ---> view users list sorted by Name first, then by id if necessary.


Testing -
1. run the specfile at app/helpers/auth/sorter.spec.ts

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
<img width="1044" alt="Screen Shot 2019-09-17 at 3 46 26 PM" src="https://user-images.githubusercontent.com/16737484/65085222-61a90500-d962-11e9-863f-4fc454bd12df.png">
